data/src/main/java/com/knocklock/data/repository/UserRepositoryImpl.kt

```diff
@@ -2,21 +2,33 @@ package com.knocklock.data.repository
 
 import androidx.datastore.core.DataStore
 import com.knocklock.data.mapper.toDomain
+import com.knocklock.data.source.local.userpreference.AuthenticationType
 import com.knocklock.data.source.local.userpreference.UserPreference
-import com.knocklock.domain.model.User
 import com.knocklock.domain.repository.UserRepository
-import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
 import javax.inject.Inject
 
 class UserRepositoryImpl @Inject constructor(
-    private val userDataStore: DataStore<UserPreference>
+    private val userDataStore: DataStore<UserPreference>,
 ) : UserRepository {
-    override suspend fun getUser(): User =
-        userDataStore.data.first().toDomain()
+
+    override fun getUser() = userDataStore.data.map {
+        it.toDomain()
+    }
 
     override suspend fun updatedPassword(password: String) {
         userDataStore.updateData { userPreference ->
             userPreference.copy(password = password)
         }
     }
+
+    override suspend fun changeMode(isPasswordMode: Boolean) {
+        userDataStore.updateData { userPreference ->
+            if (isPasswordMode) {
+                userPreference.copy(authenticationType = AuthenticationType.PASSWORD)
+            } else {
+                userPreference.copy(authenticationType = AuthenticationType.GESTURE)
+            }
+        }
+    }
```

Comment:
추후 Auth 타입 추가되는 것을 고려해서 `AuthenticationType`을 넘기는 것이 좋을 것 같은데 어떻게 생각하시나요?
이렇게 되면 domain -> data model로 변경되는 mapper 함수가 필요하겠네요 🧑‍🦱 

```suggestion
    override suspend fun changeMode(authenticationType: AuthenticationType) {
        userDataStore.updateData { userPreference ->
              userPreference.copy(authenticationType = authenticationType)
        }
    }
```

Reply:
현재 비밀번호 유무만 존재하기 때문에 boolean 값을 넘겨줬는데 추후에 pin이나 패턴같은게 추가되는걸 고려하자는 거겠죠??

Reply:
맞아요!